ImageTaggingApp.Tests/ImageTagsSavingToDatabaseServiceTests_Unit.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using ImageTaggingApp.Console.App.Entities;
+using ImageTaggingApp.Console.App.Services;
+using NUnit.Framework;
+using Raven.Client;
+using Raven.Client.Embedded;
+
+namespace ImageTaggingApp.Tests {
+    [TestFixture]
+    [Category(TestCategory.UNIT)]
+    public class ImageTagsSavingToDatabaseServiceTests_Unit {
+        private IDocumentStore _store;
+
+        [SetUp]
+        public void EachTestSetup() {
+            _store = new EmbeddableDocumentStore {
+                RunInMemory = true
+            }.Initialize();
+        }
+
+        [Test]
+        public void ValidateTaggedImagesAreSaved() {
+            ImageTagsSavingToDatabaseService service = new ImageTagsSavingToDatabaseService(_store);
+            var taggedImagesExpected = new List<Image>();
+            var random = new Random();
+            for (int i = 0; i < 20; i++) {
+                var image = new Image {
+                    Path = "ImagesPath" + i,
+                    ImageMetadata = new ImageMetadata {
+                        Tags = new List<Tag> {
+                            new Tag {Name = "Name1_" + i, Probability = random.NextDouble()},
```

Comment:
Using random in unit tests is a bad idea, even if it would not hurt in this place, it is a bad habbit (F.I.R.S.T Principles of Unit Testing - unit tet should be repeatable, it ain't with random :)